● To check whether there is any increase in sales after
stepping into digital marketing.


In [1]:
#Ho : there is no increase in sales after stepping into digital marketing.
#H1 : there is increase in sales after stepping into digital marketing.

In [18]:
import pandas as pd

In [19]:
sales = pd.read_csv("Sales_add.csv")

In [20]:
sales

Month      Region      Manager  Sales_before_digital_add(in $)  \
0    Month-1  Region - A  Manager - A                          132921   
1    Month-2  Region - A  Manager - C                          149559   
2    Month-3  Region - B  Manager - A                          146278   
3    Month-4  Region - B  Manager - B                          152167   
4    Month-5  Region - C  Manager - B                          159525   
5    Month-6  Region - A  Manager - B                          137163   
6    Month-7  Region - C  Manager - C                          130625   
7    Month-8  Region - A  Manager - A                          131140   
8    Month-9  Region - B  Manager - C                          171259   
9   Month-10  Region - C  Manager - B                          141956   
10  Month-11  Region - B  Manager - C                          159339   
11  Month-12  Region - A  Manager - A                          178939   
12  Month-13  Region - B  Manager - A                          145062   
13  Month-14  Region - A  Manager - A                          151514   
14  Month-15  Region - C  Manager - A                          147463   
15  Month-16  Region - A  Manager - B                          177195   
16  Month-17  Region - A  Manager - C                          140862   
17  Month-18  Region - C  Manager - B                          167996   
18  Month-19  Region - B  Manager - A                          132135   
19  Month-20  Region - A  Manager - B                          152493   
20  Month-21  Region - B  Manager - A                          147425   
21  Month-22  Region - A  Manager - C                          130263   

    Sales_After_digital_add(in $)  
0                          270390  
1                          223334  
2                          244243  
3                          231808  
4                          258402  
5                          256948  
6                          222106  
7                          230637  
8                          226261  
9                          193735  
10                         203175  
11                         276279  
12                         205180  
13                         253131  
14                         229336  
15                         187305  
16                         234908  
17                         191517  
18                         227040  
19                         212579  
20                         263388  
21                         243020

In [21]:
from scipy.stats import ttest_rel

In [22]:
ttest_rel(sales["Sales_before_digital_add(in $)"],sales["Sales_After_digital_add(in $)"])

Ttest_relResult(statistic=-12.09070525287017, pvalue=6.336667004575778e-11)

In [23]:
from researchpy import ttest as rpTtest

In [24]:
rpRes = rpTtest(sales["Sales_before_digital_add(in $)"],sales["Sales_After_digital_add(in $)"],equal_variances = True,paired = True)
rpRes

(                         Variable     N           Mean            SD  \
 0  Sales_before_digital_add(in $)  22.0  149239.954545  14844.042921   
 1   Sales_After_digital_add(in $)  22.0  231123.727273  25556.777061   
 2                            diff  22.0  -81883.772727  31765.635671   
 
             SE      95% Conf.       Interval  
 0  3164.760584  142658.474619  155821.434472  
 1  5448.723177  219792.487118  242454.967428  
 2  6772.456281  -95967.866571  -67799.678883  ,
                                Paired samples t-test     results
 0  Difference (Sales_before_digital_add(in $) - S... -81883.7727
 1                              Degrees of freedom =      21.0000
 2                                               t =     -12.0907
 3                           Two side test p value =       0.0000
 4                          Difference < 0 p value =       0.0000
 5                          Difference > 0 p value =       1.0000
 6                                       Cohen's d 

|T| = 12.0907  and P-Value = 6.336667004575778     
Since P-Value < T-Value we reject null Hypothesis ie there is increase in sales after stepping into digital marketing.

● The company needs to check whether there is any dependency between the
features “Region” and “Manager”.

In [25]:
a = sales["Region"]
b = sales["Manager"]

In [26]:
dep = pd.crosstab(a,b)
dep


Manager     Manager - A  Manager - B  Manager - C
Region                                           
Region - A            4            3            3
Region - B            4            1            2
Region - C            1            3            1

In [28]:
from scipy.stats import chi2_contingency

In [29]:
chiVal, pVal, df, exp = chi2_contingency(dep)
chiVal,pVal,df,exp

(3.050566893424036,
 0.5493991051158094,
 4,
 array([[4.09090909, 3.18181818, 2.72727273],
        [2.86363636, 2.22727273, 1.90909091],
        [2.04545455, 1.59090909, 1.36363636]]))

Chi2 Value = 3.050566893424036 :  P-Value = 0.5493991051158094  : df = 4
        
        Since P-Value > 0.05 , there is no dependency between the features “Region” and “Manager”

Here Company has focus more on Digital Marketing Side to increase the Sales